### ISSUE TITLE
CI should test latest Python (stable + alpha) again

### ISSUE THREAD

**Issue by wisp3rwind** [beetbox/beets#5450]
At some point during CI workflow refactoring, tests for the most recent Python versions were removed. These are essential to check for deprecated features in the language and standard library.

Currently, this affects for example #5420 (the issue is not being observed by CI) and #5235 (the change is effectively not tested due to a version check).

**Comment by snejus** [comment]
I removed support for the latest version [here](https://github.com/beetbox/beets/issues/5450) since I found that the tests failed (for some reason). I thought that issue could be addressed in a separate PR where support could be added back.

**Comment by wisp3rwind** [comment]
Sure, no problem with that! I felt it would be good to have an issue to keep track of that, though :)

**Comment by semohr** [comment]
I don't think we should test pre-release Python versions. We support and test all stable Python releases, and I think that's the right scope for this project. I'd consider this issue resolved.
